CoreUtilities/src/au/gov/asd/tac/constellation/utilities/genericjsonio/JsonIO.java

```diff
@@ -317,6 +444,63 @@ public static <T> T loadJsonPreferences(final Optional<String> loadDir, final Ty
         return loadJsonPreferences(loadDir, Optional.empty(), expectedFormat, OBJECT_MAPPER);
     }
 
+    public static <T> T loadJsonPreferencesWithFilePrefix(final Optional<String> loadDir, final Optional<String> filePrefix,
+            final TypeReference<T> expectedFormat) {
+
+        return loadJsonPreferencesForFile(loadDir, filePrefix, file -> {
+            try {
+                return OBJECT_MAPPER.readValue(file, expectedFormat);
+            } catch (final IOException ioe) {
+                LOGGER.log(Level.WARNING, String.format(FILE_READ_ERROR, file.getName()), ioe);               
+            }
+            return null;
+        });
+
+    }
+
+    protected static <T> T loadJsonPreferencesForFile(final Optional<String> loadDir,
+            final Optional<String> filePrefix,
+            final Function<File, T> deserializationFunction) {
+        final File preferenceDirectory = getPrefereceFileDirectory(loadDir);
+
+        // List the files in the supplied directory that have the required file extension
+        // and if filePrefix was supplied, start with the provided prefix.
+        final String[] names;
+        if (preferenceDirectory.isDirectory()) {
+            names = preferenceDirectory.list((File dir, String name)
```

Comment:
add finals to lambda types or remove explicit type mentions

Reply:
done